velox/substrait/SubstraitParser.cpp

```diff
@@ -60,6 +60,11 @@ std::shared_ptr<SubstraitParser::SubstraitType> SubstraitParser::parseType(
       nullability = substraitType.string().nullability();
       break;
     }
+    case ::substrait::Type::KindCase::kUserDefined: {
+      typeName = "UNKNOWN";
```

Comment:
This code is confusing, because it reads as "map any user-defined type to UNKNOWN". Perhaps, add a check here for the type name and remove checkTypeExtension.

Reply:
I think it's better to add check at the begin of the plan convertor, since we don't support other typeExtension except "UNKNOWN", we should throw error when there have other type extensions and stop the convertor. And maybe we could add some comments here to do some explanation again. Your thoughts?

Reply:
Failing early is a good practice. Maybe we can do both: check and fail early + check again here.

Reply:
Done.